services/lfs/server.go

```diff
@@ -495,25 +496,44 @@ func authenticate(ctx *context.Context, repository *repo_model.Repository, autho
 		accessMode = perm.AccessModeWrite
 	}
 
-	// ctx.IsSigned is unnecessary here, this will be checked in perm.CanAccess
-	perm, err := access_model.GetUserRepoPermission(ctx, repository, ctx.Doer)
-	if err != nil {
-		log.Error("Unable to GetUserRepoPermission for user %-v in repo %-v Error: %v", ctx.Doer, repository)
-		return false
-	}
+	if ctx.Data["IsActionsToken"] == true {
```

Comment:
```suggestion
	if ctx.Doer.IsActions() {
```

Reply:
emmm, we need to check `ctx.Doer == nill` if use `ctx.Doer.IsActions()`
maybe the old code is better.